### PR TITLE
fix(preset-attributify): filter invalid extraction

### LINKED ISSUE #2037: upgrade from 0.46.5 to 0.47.0 breaks a react project
Upgrade from 0.46.5 to 0.47.0 breaks my build (a react SPA project). Below is the diff before and after, running unocss cli:

Screenshot 2022-12-31 at 20 28 25

and the only places with the word container:

Screenshot 2022-12-31 at 20 29 31

and my unocss.config.ts file:

Screenshot 2022-12-31 at 20 34 41

### PR DESCRIPTION
**Pull request by sibbng**
Fixes #2037

### CODE REVIEW

**Comment by zyyv** [comment]
Simple and Violent😅, but it can solve the problem temporarily. I think the best way is to update the reg.

File: packages/preset-attributify/src/extractor.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const extractorAttributify = (options?: AttributifyOptions): Extractor =>
           }
 
           if (!content) {
-            if (isValidSelector(name) && nonValuedAttribute !== false) {
+            if (isValidSelector(name) && nonValuedAttribute !== false && !name.endsWith(':')) {
```
Comment (antfu) [inline packages/preset-attributify/src/extractor.ts:40]: Would this be too specific? Should we improve the rules to not handle those cases instead?

Comment (sibbng) [reply packages/preset-attributify/src/extractor.ts]: Done 👍

**Comment by sibbng** [comment]
> Simple and Violent😅, but it can solve the problem temporarily. I think the best way is to update the reg.

I tried, but couldn't come up with one. In my opinion @antfu's suggestion is better. I updated the PR with his suggestion. Now we can still target attributes that end with `:`, or people who need it.

**Comment by XiNiHa** [comment]
This broke my use case ;(
Minimal repro:

**Pull request merged**